src/elysium/createpayload.cpp

```diff
@@ -672,5 +682,95 @@ std::vector<unsigned char> CreatePayload_SimpleSpend(
     return payload;
 }
 
+std::vector<unsigned char> CreatePayload_CreateLelantusMint(
+    uint32_t propertyId, lelantus::PublicCoin const &pubcoin, MintEntryId const &id,
+    uint64_t value,  std::vector<unsigned char> const &schnorrProof)
+{
+    if (schnorrProof.size() != 98) {
+        throw std::invalid_argument("Schnorr proof size is invalid");
+    }
+
+    std::vector<unsigned char> payload;
+    uint16_t messageVer = 0;
+    uint16_t messageType = ELYSIUM_TYPE_LELANTUS_MINT;
+    elysium::swapByteOrder(messageVer);
+    elysium::swapByteOrder(messageType);
+    elysium::swapByteOrder(propertyId);
+    elysium::swapByteOrder(value);
+
+    // Meta data
+    PUSH_BACK_BYTES(payload, messageVer);
+    PUSH_BACK_BYTES(payload, messageType);
+    PUSH_BACK_BYTES(payload, propertyId);
+
+    // Mint data
+    CDataStream serialized(SER_NETWORK, CLIENT_VERSION);
+    serialized << pubcoin;
+    serialized << id;
+    payload.insert(payload.end(), serialized.begin(), serialized.end());
+
+    // Additional mint data
+    PUSH_BACK_BYTES(payload, value);
+    payload.insert(payload.end(), schnorrProof.begin(), schnorrProof.end());
+
+    return payload;
+}
+
+std::vector<unsigned char> CreatePayload_CreateLelantusJoinSplit(
+    uint32_t propertyId,
+    uint64_t amount,
+    lelantus::JoinSplit const &joinSplit,
+    boost::optional<JoinSplitMint> const &mint)
+{
+    std::vector<unsigned char> payload;
+    uint16_t messageVer = 0;
+    uint16_t messageType = ELYSIUM_TYPE_LELANTUS_JOINSPLIT;
+    elysium::swapByteOrder(messageVer);
+    elysium::swapByteOrder(messageType);
+    elysium::swapByteOrder(propertyId);
+
+    // Meta data
+    PUSH_BACK_BYTES(payload, messageVer);
+    PUSH_BACK_BYTES(payload, messageType);
+    PUSH_BACK_BYTES(payload, propertyId);
+
+    // amount
+    elysium::swapByteOrder(amount);
+    PUSH_BACK_BYTES(payload, amount);
+
+    // Mint data
+    {
+        CDataStream serialized(SER_NETWORK, CLIENT_VERSION);
+        serialized << joinSplit;
+        payload.insert(payload.end(), serialized.begin(), serialized.end());
+    }
+
+    // change
+    if (mint.has_value()) {
```

Comment:
I cannot compile on this line 
`elysium/createpayload.cpp:749:14: error: ‘const class boost::optional<elysium::JoinSplitMint>’ has no member named ‘has_value’; did you mean ‘value’?
     if (mint.has_value()) {
              ^~~~~~~~~
              value`

Reply:
@thebevrishot